cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager.go

```diff
@@ -180,6 +201,49 @@ func CreateNodePoolManager(cloudConfigPath string, discoveryOpts cloudprovider.N
 	return manager, nil
 }
 
+func autoDiscoverNodeGroups(m *ociManagerImpl, okeClient okeClient, nodeGroup nodeGroupAutoDiscovery) (bool, error) {
+	var resp, reqErr = okeClient.ListNodePools(context.Background(), oke.ListNodePoolsRequest{
+		ClusterId:     common.String(nodeGroup.clusterId),
+		CompartmentId: common.String(nodeGroup.compartmentId),
+	})
+	if reqErr != nil {
+		klog.Errorf("failed to fetch the nodepool list with clusterId: %s, compartmentId: %s. Error: %v", nodeGroup.clusterId, nodeGroup.compartmentId, reqErr)
+		return false, reqErr
+	}
+	for _, nodePoolSummary := range resp.Items {
+		klog.V(5).Infof("found nodepool %v", nodePoolSummary)
+		if validateNodepoolTags(nodeGroup.tags, nodePoolSummary.FreeformTags, nodePoolSummary.DefinedTags) {
```

Comment:
There are a few types of tags including [defined tags](https://docs.oracle.com/en-us/iaas/Content/Tagging/Tasks/managingtagsandtagnamespaces_topic-Managing_Tag_Namespaces.htm) and [free form tags](https://docs.oracle.com/en-us/iaas/Content/Tagging/Concepts/understandingfreeformtags.htm).  

As I understand it, user defined tags on a Node Pool resource would appear in the form tags (i.e. `nodePoolSummary.FreeformTags` not `nodePoolSummary.DefinedTags`).  Is there a reason we're not checking all the tag namspaces for a match?

Reply:
It is not only Freeform tags. Users can also create their own namespace and defined tags. We check both of them to make sure we don't miss a tag applied by the user.

Defined tag holds a namespace but FreeForm tag does not.
- **Defined tag :** namepsace.tagKey=tagValue
- **Freeform tag:** tagKey=tagValue

When we query Nodepool through api, the response returns them in separate fields.
- FreeformTags is a map[string=>string]
- DefinedTags is a map[string => map[string]string]